File: clients/node/sdk/models/account.ts
import {createAccountFromDiscriminatorValue} from './createAccountFromDiscriminatorValue';
import {createEmojiFromDiscriminatorValue} from './createEmojiFromDiscriminatorValue';
import {createFieldFromDiscriminatorValue} from './createFieldFromDiscriminatorValue';
import {createSourceFromDiscriminatorValue} from './createSourceFromDiscriminatorValue';
import {Emoji, Field, Source} from './index';
import {AdditionalDataHolder, Parsable, ParseNode, SerializationWriter} from '@microsoft/kiota-abstractions';

export class Account implements AdditionalDataHolder, Parsable {
    /** The acct property */
    private _acct?: string | undefined;
    /** Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well. */
    private _additionalData: Record<string, unknown>;
    /** The avatar property */
    private _avatar?: string | undefined;
    /** The avatar_static property */
    private _avatar_static?: string | undefined;
    /** The bot property */
    private _bot?: boolean | undefined;
    /** The created_at property */
    private _created_at?: string | undefined;
    /** The discoverable property */
    private _discoverable?: boolean | undefined;
    /** The display_name property */
    private _display_name?: string | undefined;
    /** The emojis property */
    private _emojis?: Emoji[] | undefined;
    /** The fields property */
    private _fields?: Field[] | undefined;
    /** The followers_count property */
    private _followers_count?: number | undefined;
    /** The following_count property */
    private _following_count?: number | undefined;
    /** The header property */
    private _header?: string | undefined;
    /** The header_static property */
    private _header_static?: string | undefined;
    /** The id property */
    private _id?: string | undefined;
    /** The last_status_at property */
    private _last_status_at?: string | undefined;
    /** The locked property */
    private _locked?: boolean | undefined;
    /** The moved property */
    private _moved?: Account | undefined;
    /** The mute_expires_at property */
    private _mute_expires_at?: string | undefined;
    /** The note property */
    private _note?: string | undefined;
    /** The souce property */
    private _souce?: Source | undefined;
    /** The statuses_count property */
    private _statuses_count?: number | undefined;
    /** The suspended property */
    private _suspended?: boolean | undefined;
    /** The url property */
    private _url?: string | undefined;
    /** The username property */
    private _username?: string | undefined;
    /**
     * Gets the acct property value. The acct property
     * @returns a string
     */
    public get acct() {
        return this._acct;
    };
    /**
     * Sets the acct property value. The acct property
     * @param value Value to set for the acct property.
     */
    public set acct(value: string | undefined) {
        this._acct = value;
    };
    /**
     * Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @returns a Record<string, unknown>
     */
    public get additionalData() {
        return this._additionalData;
    };
    /**
     * Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param value Value to set for the AdditionalData property.
     */
    public set additionalData(value: Record<string, unknown>) {
        this._additionalData = value;
    };
    /**
     * Gets the avatar property value. The avatar property
     * @returns a string
     */
    public get avatar() {
        return this._avatar;
    };
    /**
     * Sets the avatar property value. The avatar property
     * @param value Value to set for the avatar property.
     */
    public set avatar(value: string | undefined) {
        this._avatar = value;
    };
    /**
     * Gets the avatar_static property value. The avatar_static property
     * @returns a string
     */
    public get avatar_static() {
        return this._avatar_static;
    };
    /**
     * Sets the avatar_static property value. The avatar_static property
     * @param value Value to set for the avatar_static property.
     */
    public set avatar_static(value: string | undefined) {
        this._avatar_static = value;
    };
    /**
     * Gets the bot property value. The bot property
     * @returns a boolean
     */
    public get bot() {
        return this._bot;
    };
    /**
     * Sets the bot property value. The bot property
     * @param value Value to set for the bot property.
     */
    public set bot(value: boolean | undefined) {
        this._bot = value;
    };
    /**
     * Instantiates a new Account and sets the default values.
     */
    public constructor() {
        this._additionalData = {};
    };
    /**
     * Gets the created_at property value. The created_at property
     * @returns a string
     */
    public get created_at() {
        return this._created_at;
    };
    /**
     * Sets the created_at property value. The created_at property
     * @param value Value to set for the created_at property.
     */
    public set created_at(value: string | undefined) {
        this._created_at = value;
    };
    /**
     * Gets the discoverable property value. The discoverable property
     * @returns a boolean
     */
    public get discoverable() {
        return this._discoverable;
    };
    /**
     * Sets the discoverable property value. The discoverable property
     * @param value Value to set for the discoverable property.
     */
    public set discoverable(value: boolean | undefined) {
        this._discoverable = value;
    };
    /**
     * Gets the display_name property value. The display_name property
     * @returns a string
     */
    public get display_name() {
        return this._display_name;
    };
    /**
     * Sets the display_name property value. The display_name property
     * @param value Value to set for the display_name property.
     */
    public set display_name(value: string | undefined) {
        this._display_name = value;
    };
    /**
     * Gets the emojis property value. The emojis property
     * @returns a Emoji
     */
    public get emojis() {
        return this._emojis;
    };
    /**
     * Sets the emojis property value. The emojis property
     * @param value Value to set for the emojis property.
     */
    public set emojis(value: Emoji[] | undefined) {
        this._emojis = value;
    };
    /**
     * Gets the fields property value. The fields property
     * @returns a Field
     */
    public get fields() {
        return this._fields;
    };
    /**
     * Sets the fields property value. The fields property
     * @param value Value to set for the fields property.
     */
    public set fields(value: Field[] | undefined) {
        this._fields = value;
    };
    /**
     * Gets the followers_count property value. The followers_count property
     * @returns a integer
     */
    public get followers_count() {
        return this._followers_count;
    };
    /**
     * Sets the followers_count property value. The followers_count property
     * @param value Value to set for the followers_count property.
     */
    public set followers_count(value: number | undefined) {
        this._followers_count = value;
    };
    /**
     * Gets the following_count property value. The following_count property
     * @returns a integer
     */
    public get following_count() {
        return this._following_count;
    };
    /**
     * Sets the following_count property value. The following_count property
     * @param value Value to set for the following_count property.
     */
    public set following_count(value: number | undefined) {
        this._following_count = value;
    };
    /**
     * The deserialization information for the current model
     * @returns a Record<string, (node: ParseNode) => void>
     */
    public getFieldDeserializers() : Record<string, (node: ParseNode) => void> {
        return {
            "acct": n => { this.acct = n.getStringValue(); },
            "avatar": n => { this.avatar = n.getStringValue(); },
            "avatar_static": n => { this.avatar_static = n.getStringValue(); },
            "bot": n => { this.bot = n.getBooleanValue(); },
            "created_at": n => { this.created_at = n.getStringValue(); },
            "discoverable": n => { this.discoverable = n.getBooleanValue(); },
            "display_name": n => { this.display_name = n.getStringValue(); },
            "emojis": n => { this.emojis = n.getCollectionOfObjectValues<Emoji>(createEmojiFromDiscriminatorValue); },
            "fields": n => { this.fields = n.getCollectionOfObjectValues<Field>(createFieldFromDiscriminatorValue); },
            "followers_count": n => { this.followers_count = n.getNumberValue(); },
            "following_count": n => { this.following_count = n.getNumberValue(); },
            "header": n => { this.header = n.getStringValue(); },
            "header_static": n => { this.header_static = n.getStringValue(); },
            "id": n => { this.id = n.getStringValue(); },
            "last_status_at": n => { this.last_status_at = n.getStringValue(); },
            "locked": n => { this.locked = n.getBooleanValue(); },
            "moved": n => { this.moved = n.getObjectValue<Account>(createAccountFromDiscriminatorValue); },
            "mute_expires_at": n => { this.mute_expires_at = n.getStringValue(); },
            "note": n => { this.note = n.getStringValue(); },
            "souce": n => { this.souce = n.getObjectValue<Source>(createSourceFromDiscriminatorValue); },
            "statuses_count": n => { this.statuses_count = n.getNumberValue(); },
            "suspended": n => { this.suspended = n.getBooleanValue(); },
            "url": n => { this.url = n.getStringValue(); },
            "username": n => { this.username = n.getStringValue(); },
        };
    };
    /**
     * Gets the header property value. The header property
     * @returns a string
     */
    public get header() {
        return this._header;
    };
    /**
     * Sets the header property value. The header property
     * @param value Value to set for the header property.
     */
    public set header(value: string | undefined) {
        this._header = value;
    };
    /**
     * Gets the header_static property value. The header_static property
     * @returns a string
     */
    public get header_static() {
        return this._header_static;
    };
    /**
     * Sets the header_static property value. The header_static property
     * @param value Value to set for the header_static property.
     */
    public set header_static(value: string | undefined) {
        this._header_static = value;
    };
    /**
     * Gets the id property value. The id property
     * @returns a string
     */
    public get id() {
        return this._id;
    };
    /**
     * Sets the id property value. The id property
     * @param value Value to set for the id property.
     */
    public set id(value: string | undefined) {
        this._id = value;
    };
    /**
     * Gets the last_status_at property value. The last_status_at property
     * @returns a string
     */
    public get last_status_at() {
        return this._last_status_at;
    };
    /**
     * Sets the last_status_at property value. The last_status_at property
     * @param value Value to set for the last_status_at property.
     */
    public set last_status_at(value: string | undefined) {
        this._last_status_at = value;
    };
    /**
     * Gets the locked property value. The locked property
     * @returns a boolean
     */
    public get locked() {
        return this._locked;
    };
    /**
     * Sets the locked property value. The locked property
     * @param value Value to set for the locked property.
     */
    public set locked(value: boolean | undefined) {
        this._locked = value;
    };
    /**
     * Gets the moved property value. The moved property
     * @returns a Account
     */
    public get moved() {
        return this._moved;
    };
    /**
     * Sets the moved property value. The moved property
     * @param value Value to set for the moved property.
     */
    public set moved(value: Account | undefined) {
        this._moved = value;
    };
    /**
     * Gets the mute_expires_at property value. The mute_expires_at property
     * @returns a string
     */
    public get mute_expires_at() {
        return this._mute_expires_at;
    };
    /**
     * Sets the mute_expires_at property value. The mute_expires_at property
     * @param value Value to set for the mute_expires_at property.
     */
    public set mute_expires_at(value: string | undefined) {
        this._mute_expires_at = value;
    };
    /**
     * Gets the note property value. The note property
     * @returns a string
     */
    public get note() {
        return this._note;
    };
    /**
     * Sets the note property value. The note property
     * @param value Value to set for the note property.
     */
    public set note(value: string | undefined) {
        this._note = value;
    };
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    public serialize(writer: SerializationWriter) : void {
        if(!writer) throw new Error("writer cannot be undefined");
        writer.writeStringValue("acct", this.acct);
        writer.writeStringValue("avatar", this.avatar);
        writer.writeStringValue("avatar_static", this.avatar_static);
        writer.writeBooleanValue("bot", this.bot);
        writer.writeStringValue("created_at", this.created_at);
        writer.writeBooleanValue("discoverable", this.discoverable);
        writer.writeStringValue("display_name", this.display_name);
        writer.writeCollectionOfObjectValues<Emoji>("emojis", this.emojis);
        writer.writeCollectionOfObjectValues<Field>("fields", this.fields);
        writer.writeNumberValue("followers_count", this.followers_count);
        writer.writeNumberValue("following_count", this.following_count);
        writer.writeStringValue("header", this.header);
        writer.writeStringValue("header_static", this.header_static);
        writer.writeStringValue("id", this.id);
        writer.writeStringValue("last_status_at", this.last_status_at);
        writer.writeBooleanValue("locked", this.locked);
        writer.writeObjectValue<Account>("moved", this.moved);
        writer.writeStringValue("mute_expires_at", this.mute_expires_at);
        writer.writeStringValue("note", this.note);
        writer.writeObjectValue<Source>("souce", this.souce);
        writer.writeNumberValue("statuses_count", this.statuses_count);
        writer.writeBooleanValue("suspended", this.suspended);
        writer.writeStringValue("url", this.url);
        writer.writeStringValue("username", this.username);
        writer.writeAdditionalData(this.additionalData);
    };
    /**
     * Gets the souce property value. The souce property
     * @returns a Source
     */
    public get souce() {
        return this._souce;
    };
    /**
     * Sets the souce property value. The souce property
     * @param value Value to set for the souce property.
     */
    public set souce(value: Source | undefined) {
        this._souce = value;
    };
    /**
     * Gets the statuses_count property value. The statuses_count property
     * @returns a integer
     */
    public get statuses_count() {
        return this._statuses_count;
    };
    /**
     * Sets the statuses_count property value. The statuses_count property
     * @param value Value to set for the statuses_count property.
     */
    public set statuses_count(value: number | undefined) {
        this._statuses_count = value;
    };
    /**
     * Gets the suspended property value. The suspended property
     * @returns a boolean
     */
    public get suspended() {
        return this._suspended;
    };
    /**
     * Sets the suspended property value. The suspended property
     * @param value Value to set for the suspended property.
     */
    public set suspended(value: boolean | undefined) {
        this._suspended = value;
    };
    /**
     * Gets the url property value. The url property
     * @returns a string
     */
    public get url() {
        return this._url;
    };
    /**
     * Sets the url property value. The url property
     * @param value Value to set for the url property.
     */
    public set url(value: string | undefined) {
        this._url = value;
    };
    /**
     * Gets the username property value. The username property
     * @returns a string
     */
    public get username() {
        return this._username;
    };
    /**
     * Sets the username property value. The username property
     * @param value Value to set for the username property.
     */
    public set username(value: string | undefined) {
        this._username = value;
    };
}

File: clients/python/sdk/models/account.py
from __future__ import annotations
from kiota_abstractions.serialization import AdditionalDataHolder, Parsable, ParseNode, SerializationWriter
from typing import Any, Callable, Dict, List, Optional, Union

from . import emoji, field, source

class Account(AdditionalDataHolder, Parsable):
    @property
    def acct(self,) -> Optional[str]:
        """
        Gets the acct property value. The acct property
        Returns: Optional[str]
        """
        return self._acct
    
    @acct.setter
    def acct(self,value: Optional[str] = None) -> None:
        """
        Sets the acct property value. The acct property
        Args:
            value: Value to set for the acct property.
        """
        self._acct = value
    
    @property
    def additional_data(self,) -> Dict[str, Any]:
        """
        Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        Returns: Dict[str, Any]
        """
        return self._additional_data
    
    @additional_data.setter
    def additional_data(self,value: Dict[str, Any]) -> None:
        """
        Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        Args:
            value: Value to set for the AdditionalData property.
        """
        self._additional_data = value
    
    @property
    def avatar(self,) -> Optional[str]:
        """
        Gets the avatar property value. The avatar property
        Returns: Optional[str]
        """
        return self._avatar
    
    @avatar.setter
    def avatar(self,value: Optional[str] = None) -> None:
        """
        Sets the avatar property value. The avatar property
        Args:
            value: Value to set for the avatar property.
        """
        self._avatar = value
    
    @property
    def avatar_static(self,) -> Optional[str]:
        """
        Gets the avatar_static property value. The avatar_static property
        Returns: Optional[str]
        """
        return self._avatar_static
    
    @avatar_static.setter
    def avatar_static(self,value: Optional[str] = None) -> None:
        """
        Sets the avatar_static property value. The avatar_static property
        Args:
            value: Value to set for the avatar_static property.
        """
        self._avatar_static = value
    
    @property
    def bot(self,) -> Optional[bool]:
        """
        Gets the bot property value. The bot property
        Returns: Optional[bool]
        """
        return self._bot
    
    @bot.setter
    def bot(self,value: Optional[bool] = None) -> None:
        """
        Sets the bot property value. The bot property
        Args:
            value: Value to set for the bot property.
        """
        self._bot = value
    
    def __init__(self,) -> None:
        """
        Instantiates a new Account and sets the default values.
        """
        # Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        self._additional_data: Dict[str, Any] = {}

        # The acct property
        self._acct: Optional[str] = None
        # The avatar property
        self._avatar: Optional[str] = None
        # The avatar_static property
        self._avatar_static: Optional[str] = None
        # The bot property
        self._bot: Optional[bool] = None
        # The created_at property
        self._created_at: Optional[str] = None
        # The discoverable property
        self._discoverable: Optional[bool] = None
        # The display_name property
        self._display_name: Optional[str] = None
        # The emojis property
        self._emojis: Optional[List[emoji.Emoji]] = None
        # The fields property
        self._fields: Optional[List[field.Field]] = None
        # The followers_count property
        self._followers_count: Optional[int] = None
        # The following_count property
        self._following_count: Optional[int] = None
        # The header property
        self._header: Optional[str] = None
        # The header_static property
        self._header_static: Optional[str] = None
        # The id property
        self._id: Optional[str] = None
        # The last_status_at property
        self._last_status_at: Optional[str] = None
        # The locked property
        self._locked: Optional[bool] = None
        # The moved property
        self._moved: Optional[Account] = None
        # The mute_expires_at property
        self._mute_expires_at: Optional[str] = None
        # The note property
        self._note: Optional[str] = None
        # The souce property
        self._souce: Optional[source.Source] = None
        # The statuses_count property
        self._statuses_count: Optional[int] = None
        # The suspended property
        self._suspended: Optional[bool] = None
        # The url property
        self._url: Optional[str] = None
        # The username property
        self._username: Optional[str] = None
    
    @property
    def created_at(self,) -> Optional[str]:
        """
        Gets the created_at property value. The created_at property
        Returns: Optional[str]
        """
        return self._created_at
    
    @created_at.setter
    def created_at(self,value: Optional[str] = None) -> None:
        """
        Sets the created_at property value. The created_at property
        Args:
            value: Value to set for the created_at property.
        """
        self._created_at = value
    
    @staticmethod
    def create_from_discriminator_value(parse_node: Optional[ParseNode] = None) -> Account:
        """
        Creates a new instance of the appropriate class based on discriminator value
        Args:
            parseNode: The parse node to use to read the discriminator value and create the object
        Returns: Account
        """
        if parse_node is None:
            raise Exception("parse_node cannot be undefined")
        return Account()
    
    @property
    def discoverable(self,) -> Optional[bool]:
        """
        Gets the discoverable property value. The discoverable property
        Returns: Optional[bool]
        """
        return self._discoverable
    
    @discoverable.setter
    def discoverable(self,value: Optional[bool] = None) -> None:
        """
        Sets the discoverable property value. The discoverable property
        Args:
            value: Value to set for the discoverable property.
        """
        self._discoverable = value
    
    @property
    def display_name(self,) -> Optional[str]:
        """
        Gets the display_name property value. The display_name property
        Returns: Optional[str]
        """
        return self._display_name
    
    @display_name.setter
    def display_name(self,value: Optional[str] = None) -> None:
        """
        Sets the display_name property value. The display_name property
        Args:
            value: Value to set for the display_name property.
        """
        self._display_name = value
    
    @property
    def emojis(self,) -> Optional[List[emoji.Emoji]]:
        """
        Gets the emojis property value. The emojis property
        Returns: Optional[List[emoji.Emoji]]
        """
        return self._emojis
    
    @emojis.setter
    def emojis(self,value: Optional[List[emoji.Emoji]] = None) -> None:
        """
        Sets the emojis property value. The emojis property
        Args:
            value: Value to set for the emojis property.
        """
        self._emojis = value
    
    @property
    def fields(self,) -> Optional[List[field.Field]]:
        """
        Gets the fields property value. The fields property
        Returns: Optional[List[field.Field]]
        """
        return self._fields
    
    @fields.setter
    def fields(self,value: Optional[List[field.Field]] = None) -> None:
        """
        Sets the fields property value. The fields property
        Args:
            value: Value to set for the fields property.
        """
        self._fields = value
    
    @property
    def followers_count(self,) -> Optional[int]:
        """
        Gets the followers_count property value. The followers_count property
        Returns: Optional[int]
        """
        return self._followers_count
    
    @followers_count.setter
    def followers_count(self,value: Optional[int] = None) -> None:
        """
        Sets the followers_count property value. The followers_count property
        Args:
            value: Value to set for the followers_count property.
        """
        self._followers_count = value
    
    @property
    def following_count(self,) -> Optional[int]:
        """
        Gets the following_count property value. The following_count property
        Returns: Optional[int]
        """
        return self._following_count
    
    @following_count.setter
    def following_count(self,value: Optional[int] = None) -> None:
        """
        Sets the following_count property value. The following_count property
        Args:
            value: Value to set for the following_count property.
        """
        self._following_count = value
    
    def get_field_deserializers(self,) -> Dict[str, Callable[[ParseNode], None]]:
        """
        The deserialization information for the current model
        Returns: Dict[str, Callable[[ParseNode], None]]
        """
        fields = {
            "acct": lambda n : setattr(self, 'acct', n.get_str_value()),
            "avatar": lambda n : setattr(self, 'avatar', n.get_str_value()),
            "avatar_static": lambda n : setattr(self, 'avatar_static', n.get_str_value()),
            "bot": lambda n : setattr(self, 'bot', n.get_bool_value()),
            "created_at": lambda n : setattr(self, 'created_at', n.get_str_value()),
            "discoverable": lambda n : setattr(self, 'discoverable', n.get_bool_value()),
            "display_name": lambda n : setattr(self, 'display_name', n.get_str_value()),
            "emojis": lambda n : setattr(self, 'emojis', n.get_collection_of_object_values(emoji.Emoji)),
            "fields": lambda n : setattr(self, 'fields', n.get_collection_of_object_values(field.Field)),
            "followers_count": lambda n : setattr(self, 'followers_count', n.get_int_value()),
            "following_count": lambda n : setattr(self, 'following_count', n.get_int_value()),
            "header": lambda n : setattr(self, 'header', n.get_str_value()),
            "header_static": lambda n : setattr(self, 'header_static', n.get_str_value()),
            "id": lambda n : setattr(self, 'id', n.get_str_value()),
            "last_status_at": lambda n : setattr(self, 'last_status_at', n.get_str_value()),
            "locked": lambda n : setattr(self, 'locked', n.get_bool_value()),
            "moved": lambda n : setattr(self, 'moved', n.get_object_value(Account)),
            "mute_expires_at": lambda n : setattr(self, 'mute_expires_at', n.get_str_value()),
            "note": lambda n : setattr(self, 'note', n.get_str_value()),
            "souce": lambda n : setattr(self, 'souce', n.get_object_value(source.Source)),
            "statuses_count": lambda n : setattr(self, 'statuses_count', n.get_int_value()),
            "suspended": lambda n : setattr(self, 'suspended', n.get_bool_value()),
            "url": lambda n : setattr(self, 'url', n.get_str_value()),
            "username": lambda n : setattr(self, 'username', n.get_str_value()),
        }
        return fields
    
    @property
    def header(self,) -> Optional[str]:
        """
        Gets the header property value. The header property
        Returns: Optional[str]
        """
        return self._header
    
    @header.setter
    def header(self,value: Optional[str] = None) -> None:
        """
        Sets the header property value. The header property
        Args:
            value: Value to set for the header property.
        """
        self._header = value
    
    @property
    def header_static(self,) -> Optional[str]:
        """
        Gets the header_static property value. The header_static property
        Returns: Optional[str]
        """
        return self._header_static
    
    @header_static.setter
    def header_static(self,value: Optional[str] = None) -> None:
        """
        Sets the header_static property value. The header_static property
        Args:
            value: Value to set for the header_static property.
        """
        self._header_static = value
    
    @property
    def id(self,) -> Optional[str]:
        """
        Gets the id property value. The id property
        Returns: Optional[str]
        """
        return self._id
    
    @id.setter
    def id(self,value: Optional[str] = None) -> None:
        """
        Sets the id property value. The id property
        Args:
            value: Value to set for the id property.
        """
        self._id = value
    
    @property
    def last_status_at(self,) -> Optional[str]:
        """
        Gets the last_status_at property value. The last_status_at property
        Returns: Optional[str]
        """
        return self._last_status_at
    
    @last_status_at.setter
    def last_status_at(self,value: Optional[str] = None) -> None:
        """
        Sets the last_status_at property value. The last_status_at property
        Args:
            value: Value to set for the last_status_at property.
        """
        self._last_status_at = value
    
    @property
    def locked(self,) -> Optional[bool]:
        """
        Gets the locked property value. The locked property
        Returns: Optional[bool]
        """
        return self._locked
    
    @locked.setter
    def locked(self,value: Optional[bool] = None) -> None:
        """
        Sets the locked property value. The locked property
        Args:
            value: Value to set for the locked property.
        """
        self._locked = value
    
    @property
    def moved(self,) -> Optional[Account]:
        """
        Gets the moved property value. The moved property
        Returns: Optional[Account]
        """
        return self._moved
    
    @moved.setter
    def moved(self,value: Optional[Account] = None) -> None:
        """
        Sets the moved property value. The moved property
        Args:
            value: Value to set for the moved property.
        """
        self._moved = value
    
    @property
    def mute_expires_at(self,) -> Optional[str]:
        """
        Gets the mute_expires_at property value. The mute_expires_at property
        Returns: Optional[str]
        """
        return self._mute_expires_at
    
    @mute_expires_at.setter
    def mute_expires_at(self,value: Optional[str] = None) -> None:
        """
        Sets the mute_expires_at property value. The mute_expires_at property
        Args:
            value: Value to set for the mute_expires_at property.
        """
        self._mute_expires_at = value
    
    @property
    def note(self,) -> Optional[str]:
        """
        Gets the note property value. The note property
        Returns: Optional[str]
        """
        return self._note
    
    @note.setter
    def note(self,value: Optional[str] = None) -> None:
        """
        Sets the note property value. The note property
        Args:
            value: Value to set for the note property.
        """
        self._note = value
    
    def serialize(self,writer: SerializationWriter) -> None:
        """
        Serializes information the current object
        Args:
            writer: Serialization writer to use to serialize this model
        """
        if writer is None:
            raise Exception("writer cannot be undefined")
        writer.write_str_value("acct", self.acct)
        writer.write_str_value("avatar", self.avatar)
        writer.write_str_value("avatar_static", self.avatar_static)
        writer.write_bool_value("bot", self.bot)
        writer.write_str_value("created_at", self.created_at)
        writer.write_bool_value("discoverable", self.discoverable)
        writer.write_str_value("display_name", self.display_name)
        writer.write_collection_of_object_values("emojis", self.emojis)
        writer.write_collection_of_object_values("fields", self.fields)
        writer.write_int_value("followers_count", self.followers_count)
        writer.write_int_value("following_count", self.following_count)
        writer.write_str_value("header", self.header)
        writer.write_str_value("header_static", self.header_static)
        writer.write_str_value("id", self.id)
        writer.write_str_value("last_status_at", self.last_status_at)
        writer.write_bool_value("locked", self.locked)
        writer.write_object_value("moved", self.moved)
        writer.write_str_value("mute_expires_at", self.mute_expires_at)
        writer.write_str_value("note", self.note)
        writer.write_object_value("souce", self.souce)
        writer.write_int_value("statuses_count", self.statuses_count)
        writer.write_bool_value("suspended", self.suspended)
        writer.write_str_value("url", self.url)
        writer.write_str_value("username", self.username)
        writer.write_additional_data_value(self.additional_data)
    
    @property
    def souce(self,) -> Optional[source.Source]:
        """
        Gets the souce property value. The souce property
        Returns: Optional[source.Source]
        """
        return self._souce
    
    @souce.setter
    def souce(self,value: Optional[source.Source] = None) -> None:
        """
        Sets the souce property value. The souce property
        Args:
            value: Value to set for the souce property.
        """
        self._souce = value
    
    @property
    def statuses_count(self,) -> Optional[int]:
        """
        Gets the statuses_count property value. The statuses_count property
        Returns: Optional[int]
        """
        return self._statuses_count
    
    @statuses_count.setter
    def statuses_count(self,value: Optional[int] = None) -> None:
        """
        Sets the statuses_count property value. The statuses_count property
        Args:
            value: Value to set for the statuses_count property.
        """
        self._statuses_count = value
    
    @property
    def suspended(self,) -> Optional[bool]:
        """
        Gets the suspended property value. The suspended property
        Returns: Optional[bool]
        """
        return self._suspended
    
    @suspended.setter
    def suspended(self,value: Optional[bool] = None) -> None:
        """
        Sets the suspended property value. The suspended property
        Args:
            value: Value to set for the suspended property.
        """
        self._suspended = value
    
    @property
    def url(self,) -> Optional[str]:
        """
        Gets the url property value. The url property
        Returns: Optional[str]
        """
        return self._url
    
    @url.setter
    def url(self,value: Optional[str] = None) -> None:
        """
        Sets the url property value. The url property
        Args:
            value: Value to set for the url property.
        """
        self._url = value
    
    @property
    def username(self,) -> Optional[str]:
        """
        Gets the username property value. The username property
        Returns: Optional[str]
        """
        return self._username
    
    @username.setter
    def username(self,value: Optional[str] = None) -> None:
        """
        Sets the username property value. The username property
        Args:
            value: Value to set for the username property.
        """
        self._username = value
    

